clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -492,8 +492,19 @@ private static String parseAcks(String acksString) {
         }
     }
 
+    /**
+     * @deprecated Since 2.7.0. This will be removed in a future major release.
+     */
+    @Deprecated
     public static Map<String, Object> addSerializerToConfig(Map<String, Object> configs,
                                                             Serializer<?> keySerializer, Serializer<?> valueSerializer) {
+        return appendSerializerToConfig(configs, keySerializer, valueSerializer);
+    }
+
+    static Map<String, Object> appendSerializerToConfig(
+            Map<String, Object> configs,
```

Comment:
nit: I don't feel strong about the style here, but maybe we should consider align with other functions which has the first parameter on the same line with function name.